cocotb/decorators.py

```diff
@@ -262,7 +262,8 @@ def _force_outcome(self, outcome):
         should be inlined into `abort` below, and the call in `unschedule`
         replaced with `abort(outcome.error)`.
         """
-        assert self._outcome is None
+        if self._outcome is not None:
+            raise cocotb.InternalError("Outcome already has a value, but is being set again.")
```

Comment:
Instead perhaps just add `from cocotb.scheduler import InternalError` at the top of the file?